src/mon/OSDMonitor.cc

```diff
@@ -4806,8 +4806,12 @@ bool OSDMonitor::prepare_set_flag(MonOpRequestRef op, int flag)
   ostringstream ss;
   if (pending_inc.new_flags < 0)
     pending_inc.new_flags = osdmap.get_flags();
-  pending_inc.new_flags |= flag;
-  ss << "set " << OSDMap::get_flag_string(flag);
+  if (osdmap.test_flag(flag)) {
+    ss << OSDMap::get_flag_string(flag) << " flag is already set";
```

Comment:
- After applying your suggestion , I tested `$ceph osd set` and `$ceph osd unset` commands just hangs if I try to return `false` without calling `wait_for_finished_proposal()` in `if condition` and if I try to return `false` with calling `wait_for_finished_proposal()` then also same behavior.

- To me it looks like as `$ceph osd set` and `$ceph osd unset` both  expect to return `true` with `wait_for_finished_proposal()` as I was doing.  

- As if we are setting a already set flag means condition is true in respective of this function as this is the behavior of function and we are not failing here as flag is already set and same goes for unset as in unset also success would be if `flag is not set`.